libvast/src/chunk.cpp

```diff
@@ -99,8 +99,10 @@ chunk_ptr chunk::slice(size_type start, size_type length) const {
   return make(length, data_ + start, deleter);
 }
 
-chunk::chunk(void* ptr, size_type size, deleter_type deleter) noexcept
-  : data_{reinterpret_cast<value_type*>(ptr)}, size_{size}, deleter_{deleter} {
+chunk::chunk(const void* ptr, size_type size, deleter_type deleter) noexcept
+  : data_{reinterpret_cast<pointer>(const_cast<void*>(ptr))},
```

Comment:
We can `static_cast` instead of `reinterpret_cast` here

Reply:
Nice catch!